pypowsybl/security.py

```diff
@@ -45,6 +45,81 @@ def _limit_violation_repr(self: LimitViolation) -> str:
 
 LimitViolation.__repr__ = _limit_violation_repr  # type: ignore
 
+class Parameters:  # pylint: disable=too-few-public-methods
+    """
+    Parameters for a security analysis execution.
+
+    All parameters are first read from you configuration file, then overridden with
+    the constructor arguments.
+
+    Please note that security analysis providers may not honor all parameters, according to their capabilities.
+    For example, some providers will not be able to simulate the voltage control of shunt compensators, etc.
+    The exact behaviour of some parameters may also depend on your security analysis provider.
+    Please check the documentation of your provider for that information.
+
+    .. currentmodule:: pypowsybl.security
+
+    Args:
+        provider_parameters: Define parameters linked to the security analysis provider
+            the names of the existing parameters can be found with method ``get_provider_parameters_names``
+    """
+
+    def __init__(self,
+                 load_flow_parameters: pypowsybl.loadflow.Parameters = None,
+                 flow_proportional_threshold: float = None,
+                 low_voltage_proportional_threshold: float = None,
+                 low_voltage_absolute_threshold: float = None,
+                 high_voltage_proportional_threshold: float = None,
+                 high_voltage_absolute_threshold: float = None,
+                 provider_parameters: _Dict[str, str] = None):
+        self._init_with_default_values()
+        if load_flow_parameters is not None:
+            self.load_flow_parameters = load_flow_parameters
+        if flow_proportional_threshold is not None:
+            self.flow_proportional_threshold = flow_proportional_threshold
+        if low_voltage_proportional_threshold is not None:
+            self.low_voltage_proportional_threshold = low_voltage_proportional_threshold
+        if low_voltage_absolute_threshold is not None:
+            self.low_voltage_absolute_threshold = low_voltage_absolute_threshold
+        if high_voltage_proportional_threshold is not None:
+            self.high_voltage_proportional_threshold = high_voltage_proportional_threshold
+        if high_voltage_absolute_threshold is not None:
+            self.high_voltage_absolute_threshold = high_voltage_absolute_threshold
+        if provider_parameters is not None:
+            self.provider_parameters = provider_parameters
+
+    def _init_with_default_values(self) -> None:
+        default_parameters = _pypowsybl.SecurityAnalysisParameters()
+        self.load_flow_parameters = pypowsybl.loadflow.Parameters()
```

Comment:
I think it would be a little clearer to initialize it with the content of `default_parameters.load_flow_parameters`, it will avoid to read twice the config for LF params (once when we initialize SA params, another one here when we initialize LF params).